client/components/utils/comment-plugin.js

```diff
@@ -0,0 +1,37 @@
+import {_} from 'meteor/underscore';
+import {Plugin} from 'prosemirror-state';
+import {Decoration, DecorationSet} from 'prosemirror-view';
+
+function getDecorations(doc, vueInstance) {
+  const result = [];
+  const keys = [];
+  doc.descendants((node, pos) => {
+    const mark = _.find(node.marks, (m) => {
+      return m.type.name === "highlight";
+    });
+    if (mark) {
+      keys.push(mark.attrs["data-highlight-keys"].split(","));
+      result.push(Decoration.inline(pos, pos + node.nodeSize, {class: "highlight"}));
+    }
+  });
+  vueInstance.filterComments(_.flatten(keys));
+  return DecorationSet.create(doc, result);
+}
+export const commentPlugin = (vueInstance) => {
+  return new Plugin({
+    state: {
+      init(config, {doc}) {
+        return DecorationSet.empty;
+      },
+      apply(tr, old) {
+        return tr.docChanged ? getDecorations(tr.doc, vueInstance) : old;
```

Comment:
So this means that at every new step we call getDecorations which calls Meteor method filterComments over the wire?

Reply:
Yeah, the reason is that this allow us to handle the case where a previous deleted comment is restored via Undo (Ctrl + Z) and to set versionTo in deleted comments.

Another way to do this is to check (in client or server) if the new step is an Undo Step or if it's an step that include a deleted comment. We would have to look in how to identify this with ProseMirror if we decide to go that way.

Reply:
I understand this. This should probably be all done on the server side. I opened #45 for this.